pkg/security/probe/activity_dump.go

```diff
@@ -663,15 +668,13 @@ func (ad *ActivityDump) Encode(format dump.StorageFormat) (*bytes.Buffer, error)
 
 // EncodeJSON encodes an activity dump in the JSON format
 func (ad *ActivityDump) EncodeJSON() (*bytes.Buffer, error) {
-	ad.Lock()
-	defer ad.Unlock()
-
-	msgpRaw, err := ad.MarshalMsg(nil)
+	msgpRaw, err := ad.EncodeMSGP()
```

Comment:
Is there a reason why we don't need to lock anymore ?

Reply:
`EncodeMSGP` manages the locking